pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -190,12 +190,14 @@ protected long getNumberOfEntriesInBacklog() {
 
     @Override
     protected void disableReplicatorRead() {
-        // deactivate cursor after successfully close the producer
-        this.cursor.setInactive();
+        if (this.cursor != null) {
+            // deactivate cursor after successfully close the producer
+            this.cursor.setInactive();
```

Comment:
this is not enough to prevent an NPE here.
We should sample the reference and then use the local ref

```
ManagedCursor currentCursor = this.cursor;
if (localCursor != null) {
      localCursor.setInactive();
}
```

Reply:
It shouldn't be necessary in this cases since in the `PersistentReplicator` class, the cursor field is never updated to null after it has been set by the `openCursorComplete` callback. The cursor field is null only before the cursor has been opened.